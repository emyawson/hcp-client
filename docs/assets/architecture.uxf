<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<diagram program="umlet" version="14.2">
  <zoom_level>7</zoom_level>
  <element>
    <id>UMLPackage</id>
    <coordinates>
      <x>462</x>
      <y>14</y>
      <w>273</w>
      <h>217</h>
    </coordinates>
    <panel_attributes>*Dashboards*
--

bg=white
fg=#153275
layer=-1</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>126</x>
      <y>35</y>
      <w>119</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>strip-management
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>182</x>
      <y>84</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>authentication
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>126</x>
      <y>133</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>diagnostics
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>210</x>
      <y>182</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>...
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLPackage</id>
    <coordinates>
      <x>98</x>
      <y>7</y>
      <w>273</w>
      <h>217</h>
    </coordinates>
    <panel_attributes>Domains
--

bg=white
fg=red</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>532</x>
      <y>42</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>patient-dashboard
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>532</x>
      <y>84</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>admin-dashboard
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>532</x>
      <y>126</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>hcp-dashboard
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>532</x>
      <y>168</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>...
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>Relation</id>
    <coordinates>
      <x>238</x>
      <y>28</y>
      <w>308</w>
      <h>35</h>
    </coordinates>
    <panel_attributes>lt=&lt;.
fg=#BFBFBF
</panel_attributes>
    <additional_attributes>10.0;20.0;420.0;30.0</additional_attributes>
  </element>
  <element>
    <id>Relation</id>
    <coordinates>
      <x>301</x>
      <y>49</y>
      <w>245</w>
      <h>56</h>
    </coordinates>
    <panel_attributes>lt=&lt;.
fg=#BFBFBF</panel_attributes>
    <additional_attributes>10.0;60.0;330.0;10.0</additional_attributes>
  </element>
  <element>
    <id>Relation</id>
    <coordinates>
      <x>245</x>
      <y>56</y>
      <w>301</w>
      <h>98</h>
    </coordinates>
    <panel_attributes>lt=&lt;.
fg=#BFBFBF</panel_attributes>
    <additional_attributes>10.0;120.0;410.0;10.0</additional_attributes>
  </element>
  <element>
    <id>Relation</id>
    <coordinates>
      <x>364</x>
      <y>119</y>
      <w>112</w>
      <h>28</h>
    </coordinates>
    <panel_attributes>lt=&lt;-
use</panel_attributes>
    <additional_attributes>10.0;20.0;140.0;20.0</additional_attributes>
  </element>
  <element>
    <id>UMLPackage</id>
    <coordinates>
      <x>315</x>
      <y>308</y>
      <w>273</w>
      <h>217</h>
    </coordinates>
    <panel_attributes>*Core*
--

bg=white
fg=#6382BF
layer=-2</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>329</x>
      <y>350</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>patient
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>448</x>
      <y>392</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>strips
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>350</x>
      <y>441</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>authentication
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>UMLClass</id>
    <coordinates>
      <x>448</x>
      <y>476</y>
      <w>126</w>
      <h>21</h>
    </coordinates>
    <panel_attributes>utils
fg=#38373A</panel_attributes>
    <additional_attributes/>
  </element>
  <element>
    <id>Relation</id>
    <coordinates>
      <x>343</x>
      <y>217</y>
      <w>35</w>
      <h>105</h>
    </coordinates>
    <panel_attributes>lt=&lt;-
use</panel_attributes>
    <additional_attributes>10.0;130.0;10.0;10.0</additional_attributes>
  </element>
  <element>
    <id>Relation</id>
    <coordinates>
      <x>511</x>
      <y>224</y>
      <w>35</w>
      <h>112</h>
    </coordinates>
    <panel_attributes>lt=&lt;-
use</panel_attributes>
    <additional_attributes>10.0;140.0;10.0;10.0</additional_attributes>
  </element>
</diagram>
