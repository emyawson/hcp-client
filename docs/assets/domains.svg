<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg PUBLIC '-//W3C//DTD SVG 1.0//EN'
          'http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd'>
<svg fill-opacity="1" xmlns:xlink="http://www.w3.org/1999/xlink" color-rendering="auto" color-interpolation="auto" text-rendering="auto" stroke="black" stroke-linecap="square" width="950" stroke-miterlimit="10" shape-rendering="auto" stroke-opacity="1" fill="black" stroke-dasharray="none" font-weight="normal" stroke-width="1" viewBox="280 310 950 360" height="360" xmlns="http://www.w3.org/2000/svg" font-family="'Dialog'" font-style="normal" stroke-linejoin="miter" font-size="12px" stroke-dashoffset="0" image-rendering="auto"
><!--Generated by the Batik Graphics2D SVG Generator--><defs id="genericDefs"
  /><g
  ><defs id="defs1"
    ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath1"
      ><path d="M0 0 L2147483647 0 L2147483647 2147483647 L0 2147483647 L0 0 Z"
      /></clipPath
      ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath2"
      ><path d="M0 0 L0 310 L390 310 L390 0 Z"
      /></clipPath
      ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath3"
      ><path d="M0 0 L0 30 L180 30 L180 0 Z"
      /></clipPath
      ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath4"
      ><path d="M0 0 L0 30 L170 30 L170 0 Z"
      /></clipPath
      ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath5"
      ><path d="M0 0 L0 40 L160 40 L160 0 Z"
      /></clipPath
      ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath6"
      ><path d="M0 0 L0 140 L430 140 L430 0 Z"
      /></clipPath
      ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath7"
      ><path d="M0 0 L0 80 L350 80 L350 0 Z"
      /></clipPath
      ><clipPath clipPathUnits="userSpaceOnUse" id="clipPath8"
      ><path d="M0 0 L0 50 L440 50 L440 0 Z"
      /></clipPath
    ></defs
    ><g fill="rgb(255,255,255)" fill-opacity="0.4902" transform="translate(820,340)" stroke-opacity="0.4902" stroke="rgb(255,255,255)"
    ><path d="M0.5 0.5 L156.5 0.5 L156.5 21.3174 L389 21.3174 L389 309 L0.5 309 Z" stroke="none" clip-path="url(#clipPath2)"
    /></g
    ><g fill="rgb(21,50,117)" transform="translate(820,340)" stroke="rgb(21,50,117)"
    ><path fill="none" d="M0.5 0.5 L156.5 0.5 L156.5 21.3174 L389 21.3174 L389 309 L0.5 309 Z" clip-path="url(#clipPath2)"
      /><path fill="none" d="M0.5 21.3174 L156.5 21.3174" clip-path="url(#clipPath2)"
      /><text x="5" font-size="14px" y="15.8174" clip-path="url(#clipPath2)" font-family="sans-serif" stroke="none" font-weight="bold" xml:space="preserve"
      >Dashboards</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(920,560)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(920,560)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)"
      /><text x="83" font-size="14px" y="17.8174" clip-path="url(#clipPath3)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >...</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(920,500)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(920,500)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)"
      /><text x="37" font-size="14px" y="17.8174" clip-path="url(#clipPath3)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >hcp-dashboard</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(920,440)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(920,440)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)"
      /><text x="28" font-size="14px" y="17.8174" clip-path="url(#clipPath3)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >admin-dashboard</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(920,380)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(920,380)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)"
      /><text x="26" font-size="14px" y="17.8174" clip-path="url(#clipPath3)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >patient-dashboard</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0.4902" transform="translate(300,330)" stroke-opacity="0.4902" stroke="rgb(255,255,255)"
    ><path d="M0.5 0.5 L156.5 0.5 L156.5 21.3174 L389 21.3174 L389 309 L0.5 309 Z" stroke="none" clip-path="url(#clipPath2)"
    /></g
    ><g fill="red" transform="translate(300,330)" stroke="red"
    ><path fill="none" d="M0.5 0.5 L156.5 0.5 L156.5 21.3174 L389 21.3174 L389 309 L0.5 309 Z" clip-path="url(#clipPath2)"
      /><path fill="none" d="M0.5 21.3174 L156.5 21.3174" clip-path="url(#clipPath2)"
      /><text x="5" font-size="14px" y="15.8174" clip-path="url(#clipPath2)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >Domains</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(460,580)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(460,580)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)"
      /><text x="83" font-size="14px" y="17.8174" clip-path="url(#clipPath3)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >...</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(340,510)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(340,510)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)"
      /><text x="51" font-size="14px" y="17.8174" clip-path="url(#clipPath3)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >diagnostics</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(420,440)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(420,440)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="178.5" height="28.5" y="0.5" clip-path="url(#clipPath3)"
      /><text x="41" font-size="14px" y="17.8174" clip-path="url(#clipPath3)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >authentication</text
    ></g
    ><g fill="rgb(255,255,255)" fill-opacity="0" transform="translate(340,370)" stroke-opacity="0" stroke="rgb(255,255,255)"
    ><rect x="0.5" width="168.5" height="28.5" y="0.5" clip-path="url(#clipPath4)" stroke="none"
    /></g
    ><g fill="rgb(56,55,58)" transform="translate(340,370)" stroke="rgb(56,55,58)"
    ><rect fill="none" x="0.5" width="168.5" height="28.5" y="0.5" clip-path="url(#clipPath4)"
      /><text x="21" font-size="14px" y="17.8174" clip-path="url(#clipPath4)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >strip-management</text
    ></g
    ><g transform="translate(680,490)"
    ><path fill="none" d="M11.5 20.5 L140.5 20.5" clip-path="url(#clipPath5)"
      /><path fill="none" d="M22.2583 27 L11 20.5 L22.2583 14" clip-path="url(#clipPath5)"
      /><text x="64.3086" font-size="14px" y="16" clip-path="url(#clipPath5)" font-family="sans-serif" stroke="none" xml:space="preserve"
      >use</text
    ></g
    ><g fill="rgb(191,191,191)" stroke-miterlimit="5" stroke-dasharray="8,5" transform="translate(510,400)" stroke-linecap="butt" stroke="rgb(191,191,191)"
    ><path fill="none" d="M11.4642 120.2348 L410.5 10.5" clip-path="url(#clipPath6)"
      /><path fill="none" stroke-miterlimit="10" stroke-dasharray="none" d="M23.561 123.6495 L10.9821 120.3674 L20.1139 111.1149" clip-path="url(#clipPath6)" stroke-linecap="square"
    /></g
    ><g fill="rgb(191,191,191)" stroke-miterlimit="5" stroke-dasharray="8,5" transform="translate(590,390)" stroke-linecap="butt" stroke="rgb(191,191,191)"
    ><path fill="none" d="M11.488 60.3456 L330.5 10.5" clip-path="url(#clipPath7)"
      /><path fill="none" stroke-miterlimit="10" stroke-dasharray="none" d="M23.1208 65.1069 L10.994 60.4228 L21.1139 52.2627" clip-path="url(#clipPath7)" stroke-linecap="square"
    /></g
    ><g fill="rgb(191,191,191)" stroke-miterlimit="5" stroke-dasharray="8,5" transform="translate(500,360)" stroke-linecap="butt" stroke="rgb(191,191,191)"
    ><path fill="none" d="M11.4997 20.5244 L420.5 30.5" clip-path="url(#clipPath8)"
      /><path fill="none" stroke-miterlimit="10" stroke-dasharray="none" d="M22.0963 27.2848 L10.9999 20.5122 L22.4133 14.2886" clip-path="url(#clipPath8)" stroke-linecap="square"
    /></g
  ></g
></svg
>
