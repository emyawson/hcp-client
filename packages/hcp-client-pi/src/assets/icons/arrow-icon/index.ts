export { ArrowIcon } from './arrow.component';
export { ArrowRightIcon } from './arrow-right.component';
