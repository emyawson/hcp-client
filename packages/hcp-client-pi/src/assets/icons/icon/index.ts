export { SvgIcon, SvgIconProps } from './icon.component';
