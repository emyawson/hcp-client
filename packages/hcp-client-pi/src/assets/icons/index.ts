export { ArrowIcon, ArrowRightIcon } from './arrow-icon';
export { GearIcon } from './gear-icon/gear-icon.component';
