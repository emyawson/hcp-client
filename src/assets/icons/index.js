export * from './app-logo';
export * from './apple-fullcolor-icon';
export * from './apple-icon';
export * from './appleeaten-fullcolor-icon';
export * from './appleeaten-icon';
export * from './arrow-icon';
export * from './asterisk-icon';
export * from './basal-plus-bolus-icon';
export * from './bell-icon';
export * from './bolus-extended-icon';
export * from './bolus-multiwave-icon';
export * from './bolus-quick-icon';
export * from './bolus-standard-icon';
export * from './box-graph-icon';
export * from './ce-icon';
export * from './checkmark-icon';
export * from './circle-mark-icon';
export * from './connecting-lines-icon';
export * from './create-patient-icon';
export * from './cross-mark-icon';
export * from './date-range-icon';
export * from './disclaimer-icon';
export * from './double-ellipse-icon';
export * from './dtc-icon';
export * from './factory-icon';
export * from './flag-icon';
export * from './gear-icon';
export * from './grid-icon';
export * from './help-icon';
export * from './home-icon';
export * from './hypo-icon';
export * from './hypo-limit-icon';
export * from './hypo-symptoms-icon';
export * from './icon';
export * from './left-chevron-icon';
export * from './left-guillemet-icon';
export * from './light-bulb-icon';
export * from './line-graph-icon';
export {
  LogbookDiaryIcon,
  LogbookIcon,
  LogbookStatsIcon,
} from './logbook-icon';
export * from './logo';
export * from './maximize-icon';
export * from './mean-blood-glucose-icon';
export * from './medical-icon';
export * from './menu-icon';
export * from './minimize-icon';
export * from './minus-icon';
export * from './modified-bolus-icon';
export * from './night-icon';
export * from './no-information-icon';
export * from './overnight-icon';
export * from './patients-icon';
export * from './patient-dashboard-icon';
export * from './pencil-icon';
export * from './plus-icon';
export * from './pump-pause-icon';
export * from './pump-power-down-icon';
export * from './pump-power-up-icon';
export * from './pump-profile-change-icon';
export * from './pump-basal-rate-change-icon';
export * from './pump-run-icon';
export * from './pump-stop-icon';
export * from './pump-tbrdec-icon';
export * from './pump-tbrenddec-icon';
export * from './pump-tbrendinc-icon';
export * from './pump-tbrinc-icon';
export * from './question-icon';
export * from './rectangle-mark-icon';
export * from './right-chevron-icon';
export * from './right-guillemet-icon';
export * from './speech-bubble-icon';
export * from './stripstock-icon';
export * from './t-icon';
export * from './trash-icon';
export * from './target-range-icon';
export * from './white-line-icon';
export * from './x-icon';
export * from './patterns-icon';
export * from './device-icon';
export * from './device-error-icon';
export * from './device-assignment-error-icon';
export * from './device-download-icon';
export * from './patient-silhouette-icon';
export * from './phone-icon';
export * from './email-icon';
export * from './udtc-icon';
