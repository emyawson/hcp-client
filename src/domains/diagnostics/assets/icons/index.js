export { AppleIcon } from './apple-icon';
export { AppleEatenIcon } from './appleeaten-icon';
export { AppleEatenFullColorIcon } from './appleeaten-fullcolor-icon';
export { AppleFullColorIcon } from './apple-fullcolor-icon';
export { ArrowIcon } from './arrow-icon';
export { AsteriskIcon } from './asterisk-icon';
export { BasalPlusBolusIcon } from './basal-plus-bolus-icon';
export { BolusExtendedIcon } from './bolus-extended-icon';
export { BolusMultiwaveIcon } from './bolus-multiwave-icon';
export { BolusQuickIcon } from './bolus-quick-icon';
export { BolusStandardIcon } from './bolus-standard-icon';
export { BoxGraphIcon } from './box-graph-icon';
export { CheckmarkIcon } from './checkmark-icon';
export { CircleMarkIcon } from './circle-mark-icon';
export { ConnectingLinesIcon } from './connecting-lines-icon';
export { CrossMarkIcon } from './cross-mark-icon';
export { DateRangeIcon } from './date-range-icon';
export { DisclaimerIcon } from './disclaimer-icon';
export { DoubleEllipseIcon } from './double-ellipse-icon';
export { GearIcon } from './gear-icon';
export { GridIcon } from './grid-icon';
export { HypoIcon } from './hypo-icon';
export { HypoLimitIcon } from './hypo-limit-icon';
export { HypoSymptomsIcon } from './hypo-symptoms-icon';
export { LeftGuillemetIcon } from './left-guillemet-icon';
export { LeftChevronIcon } from './left-chevron-icon';
export { LightBulbIcon } from './light-bulb-icon';
export { LineGraphIcon } from './line-graph-icon';
export {
  LogbookIcon,
  LogbookDiaryIcon,
  LogbookStatsIcon,
} from './logbook-icon';
export { MinimizeIcon } from './minimize-icon';
export { MaximizeIcon } from './maximize-icon';
export { MeanBloodGlucoseIcon } from './mean-blood-glucose-icon';
export { MenuIcon } from './menu-icon';
export { ModifiedBolusIcon } from './modified-bolus-icon';
export { NightIcon } from './night-icon';
export { OvernightIcon } from './overnight-icon';
export { PumpBasalRateChangeIcon } from './pump-basal-rate-change-icon';
export { PumpPowerDownIcon } from './pump-power-down-icon';
export { PumpPowerUpIcon } from './pump-power-up-icon';
export { PumpProfileChangeIcon } from './pump-profile-change-icon';
export { PumpPauseIcon } from './pump-pause-icon';
export { PumpRunIcon } from './pump-run-icon';
export { PumpStopIcon } from './pump-stop-icon';
export { PumpTbrdecIcon } from './pump-tbrdec-icon';
export { PumpTbrEndDecIcon } from './pump-tbrenddec-icon';
export { PumpTbrincIcon } from './pump-tbrinc-icon';
export { PumpTbrEndIncIcon } from './pump-tbrendinc-icon';
export { RectangleMarkIcon } from './rectangle-mark-icon';
export { RightChevronIcon } from './right-chevron-icon';
export { RightGuillemetIcon } from './right-guillemet-icon';
export { TIcon } from './t-icon';
export { SnackIcon } from './snack-icon';
export { PillsIcon } from './pills-icon';
export { BedIcon } from './bed-icon';
export { RunningIcon } from './running-icon';
export { BicycleIcon } from './bicycle-icon';
export { MealIcon } from './meal-icon';
export { HyperWarningIcon } from './hyper-warning-icon';
export { HypoWarningIcon } from './hypo-warning-icon';
export { PremenstrualIcon } from './premenstrual-icon';
export { CustomIcon } from './custom-icon';
export { UserDefinedAsteriskIcon } from './user-defined-asterisk-icon';
export { StressIcon } from './stress-icon';
export { IllnessIcon } from './illness-icon';
export { FastingIcon } from './fasting-icon';
export {
  BeforeExerciseIcon,
  DuringExerciseIcon,
  AfterExerciseIcon,
} from './exercise-icon';
