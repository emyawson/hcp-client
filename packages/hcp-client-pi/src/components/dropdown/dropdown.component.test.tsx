import { shallow } from 'enzyme';
import * as React from 'react';

import { ArrowIcon } from 'src/assets/icons';

import {
  Arrow,
  createValidators,
  Dropdown,
  selectionHasValue,
} from './dropdown.component';

describe('dropdown test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(
      <Dropdown
        label="Drop down"
        placeholder="The placeholder"
        modelPath=".dropdown"
        options={[
          { label: 'Uno', value: 1 },
          { label: 'Dos', value: 2 },
          { label: 'Tres', value: 3 },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('creates RRF validators if the input is required', () => {
    expect(createValidators(true).isRequired).toBeDefined();
    expect(createValidators(false)).toEqual({});
  });
  it('validates selected option when required', () => {
    expect(selectionHasValue('test')).toBeTruthy();
    expect(selectionHasValue(null)).toBeFalsy();
    expect(selectionHasValue(1)).toBeTruthy();
    expect(selectionHasValue(undefined)).toBeFalsy();
  });
});
