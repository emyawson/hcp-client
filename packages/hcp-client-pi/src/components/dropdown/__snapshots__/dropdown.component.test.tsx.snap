// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(DropdownComponent)
    label="Drop down"
    modelPath=".dropdown"
    options={
      Array [
        Object {
          "label": "Uno",
          "value": 1,
        },
        Object {
          "label": "Dos",
          "value": 2,
        },
        Object {
          "label": "Tres",
          "value": 3,
        },
      ]
    }
    placeholder="The placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "arrowIconHeight": 8,
      "autosize": false,
      "backspaceRemoves": false,
      "borderless": true,
      "clearable": false,
      "disabled": false,
      "id": "",
      "label": "Drop down",
      "modelPath": ".dropdown",
      "multi": false,
      "onChange": [Function],
      "optionFontSize": "0.875rem",
      "options": Array [
        Object {
          "label": "Uno",
          "value": 1,
        },
        Object {
          "label": "Dos",
          "value": 2,
        },
        Object {
          "label": "Tres",
          "value": 3,
        },
      ],
      "placeholder": "The placeholder",
      "required": false,
      "searchable": true,
      "selectHeight": "3rem",
      "selectedOptionPadding": "1rem",
      "simpleValue": true,
      "theme": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "arrowIconHeight": 8,
        "autosize": false,
        "backspaceRemoves": false,
        "borderless": true,
        "clearable": false,
        "disabled": false,
        "id": "",
        "label": "Drop down",
        "modelPath": ".dropdown",
        "multi": false,
        "onChange": [Function],
        "optionFontSize": "0.875rem",
        "options": Array [
          Object {
            "label": "Uno",
            "value": 1,
          },
          Object {
            "label": "Dos",
            "value": 2,
          },
          Object {
            "label": "Tres",
            "value": 3,
          },
        ],
        "placeholder": "The placeholder",
        "required": false,
        "searchable": true,
        "selectHeight": "3rem",
        "selectedOptionPadding": "1rem",
        "simpleValue": true,
        "theme": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
